modules/core/arrow-core-data/src/test/kotlin/arrow/core/AndThenTest.kt

```diff
@@ -119,9 +122,24 @@ private fun <A> AndThen.Companion.eqK() = object : EqK<AndThenPartialOf<A>> {
     }
 }
 
+private fun AndThen.Companion.eq2K() = object : Eq2K<ForAndThen> {
+  override fun <A, B> Kind2<ForAndThen, A, B>.eqK(other: Kind2<ForAndThen, A, B>, EQA: Eq<A>, EQB: Eq<B>): Boolean =
+    (this.fix() to other.fix()).let { (ls, rs) ->
+      EQB.run {
+        ls(1).eqv(rs(1))
```

Comment:
How does this even typecheck? o.O Shouldn't the input be `A` and not `Int`?

Reply:
i just adapted the existing code 😁 isn't 'A' just resolved to 'Int' here?